pilot/pkg/networking/core/v1alpha3/listener.go

```diff
@@ -687,6 +693,15 @@ func buildHTTPConnectionManager(env model.Environment, httpOpts *httpListenerOpt
 	if env.Mesh.EnableTracing {
 		connectionManager.Tracing = &http_conn.HttpConnectionManager_Tracing{
 			OperationName: httpOpts.direction,
+			ClientSampling: &envoy_type1.Percent{
+				Value: 100.0,
+			},
+			RandomSampling: &envoy_type1.Percent{
+				Value: getTraceSampling(),
```

Comment:
Should we only apply random sampling on gateway? E.g. at gateways:
`client_sampling: 0, random_sampling: getTraceSampling(), overall_sampling: 100`
and on sidecars:
`client_sampling: 100, random_sampling: 0, overall_sampling: 100`

I think using just random sampling has an unintended effect of ever-increasing chance of sampling as request chains get longer.

Reply:
Would this prevent traffic that is generated within the mesh from being randomly traced?

Reply:
Actually, may be we can random sample on outbound sidecars + gateways only? That should take care of requests originating within the mesh.

Reply:
For requests that are initiated inside the mesh, or requests that end up directly at a service pod bypassing gateways, we should have random sampling on everywhere.
The chance of an unsampled request being sampled increases as the chain length goes up, but we don't want to preclude sampling everywhere.  `1 - (1- random_sampling ) ** length`

Reply:
OK, it's better than nothing.